cmd/kubernetes-admission-controller/extractor/from_job_test.go

```diff
@@ -0,0 +1,53 @@
+package extractor
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/api/admission/v1beta1"
+	batchV1 "k8s.io/api/batch/v1"
+	v1 "k8s.io/api/core/v1"
+)
+
+func TestFromJob(t *testing.T) {
+	// arrange
+	job := batchV1.Job{
+		ObjectMeta: testDeploymentObjectMeta,
+		Spec:       testJobSpec,
+	}
+	request := mockAdmissionRequestFromJob(t, job)
+
+	// act
+	actualMeta, actualPodSpecs, err := fromJob(request)
+
+	// assert
+	assert.NoError(t, err)
+	assert.EqualValues(t, testDeploymentObjectMeta, actualMeta)
+	assert.Contains(t, actualPodSpecs, testPodSpec)
+}
+
+func mockAdmissionRequestFromJob(t *testing.T, job batchV1.Job) v1beta1.AdmissionRequest {
+	t.Helper()
+
+	return mockAdmissionRequestFromObject(t, "Deployment", "deployments", job)
```

Comment:
Gah, I missed this, too. 😭  This should be specifying "job" values for the resource kind.

Reply:
Btw, I'm taking this as feedback that it needs to be easier to mock resources in the test code. I'll strive to improve that in future PRs.